tree/tree/src/InternalTreeUtils.cxx

```diff
@@ -403,6 +408,56 @@ std::vector<std::unique_ptr<TChain>> MakeFriends(const ROOT::TreeUtils::RFriendI
    return friends;
 }
 
+std::vector<std::string> ExpandGlob(TString glob)
+{
+   TString directory;
+
+   // TODO: Also implement this pattern in some other free function
+   // This is practically Python's os.path.split
+   Int_t slashpos = glob.Last('/');
+   if (slashpos >= 0) {
+      directory = glob(0, slashpos); // Copy the directory name
+      glob.Remove(0, slashpos + 1);  // and remove it from glob
+   } else {
+      directory = gSystem->UnixPathName(gSystem->WorkingDirectory());
+   }
+
+   const char *epath = gSystem->ExpandPathName(directory.Data());
+   void *dir = gSystem->OpenDirectory(epath);
+   delete[] epath;
+
+   if (dir) {
+      // create a TList to store the file names (not yet sorted)
+      TList l;
+      TRegexp re(glob, true);
+      const char *file;
+      TString s;
+      while ((file = gSystem->GetDirEntry(dir))) {
+         if (!strcmp(file, ".") || !strcmp(file, ".."))
+            continue;
+         s = file;
+         if ((glob != file) && s.Index(re) == kNPOS)
+            continue;
+         // Using "/" as separator here as it was done in TChain::Add
+         // In principle this should be using the appropriate platform separator
+         l.Add(new TObjString((directory + "/" + file).Data()));
+      }
+      gSystem->FreeDirectory(dir);
+      // sort the files in alphanumeric order
+      l.Sort();
+
+      // Convert TList<TString> to std::vector<std::string>
+      std::vector<std::string> ret;
+      ret.reserve(l.GetEntries());
+      for (const auto *tobjstr : ROOT::RangeStaticCast<const TObjString *>(l)) {
+         ret.push_back(tobjstr->GetName());
+      }
+      return ret;
+   } else {
+      throw std::runtime_error("Could not open directory '" + directory + "'.");
```

Comment:
I know we don't do it consistently, but it's useful to put the context of the error in the exception message, saves some work to the person that will have to debug this:

```suggestion
      throw std::runtime_error("ExpandGlob: could not open directory '" + directory + "'.");
```